src/access.rs

```diff
@@ -1,83 +1,108 @@
 //! Marker types for limiting access.
 
-/// Private trait that is implemented for the types in this module.
-pub trait Access: Copy + Default {
-    /// Ensures that this trait cannot be implemented outside of this crate.
-    #[doc(hidden)]
-    fn _private() -> _Private {
-        _Private
-    }
+/// A trait for restricting an [`Access`] type to anther [`Access`] type.
+///
+/// Restricting `Self` to `To` results in [`Self::Restricted`].
+/// Restriction is a symmetric operation.
+///
+/// The following table holds:
+///
+/// | `Self`        | `To`          | `Self` restricted to `To` |
+/// | ------------- | ------------- | ------------------------- |
+/// | `T`           | `T`           | `T`                       |
+/// | [`ReadWrite`] | `T`           | `T`                       |
+/// | [`NoAccess`]  | `T`           | [`NoAccess`]              |
+/// | [`ReadOnly`]  | [`WriteOnly`] | [`NoAccess`]              |
```

Comment:
I like the table! Maybe we should add rows for `T + NoAccess` and `WriteOnly + ReadOnly` for completeness.

Reply:
Then we'd also need `T` ∩ `ReadWrite`, but this and all the other cases are already covered by the symmetry of the operation. I fear adding these three cases would reduce clarity.

What do you think of the following proposal of highlighting the symmetry more?
I can remove the ∩ symbol again if you'd like, but it might help to highlight the symmetry.

---

# Trait volatile::access::RestrictAccess

A trait for restricting one `Access` type to another `Access` type.

Restricting `Self` to `To` results in `Self::Restricted`.

Restriction is a symmetric operation which is denoted by ∩, as it is the intersection of permissions.
The following table holds:

| `Self`        | `To`          | `Self` ∩ `To` |
| ------------- | ------------- | ------------- |
| `T`           | `T`           | `T`           |
| `ReadWrite`   | `T`           | `T`           |
| `NoAccess`    | `T`           | `NoAccess`    |
| `ReadOnly`    | `WriteOnly`   | `NoAccess`    |

Reply:
Looks good, thanks!